java/semantickernel-api/src/main/java/com/microsoft/semantickernel/hooks/KernelHook.java

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.semantickernel.hooks;
+
+public interface KernelHook {
+
+    default <T extends HookEvent> T dispatch(T event) {
```

Comment:
All these checks happen every time `executeHooks` is called, could we avoid it by having two different maps, one for pre and another for post? That way the user is also able to be explicit about what hook they're adding to the kernel by adding `Kernel.addInvokingHook` and `Kernel.addInvokedHook` methods to the API.

Reply:
this style of dispatch has been removed in the latest pr